frontend/libs/studio-components/src/components/StudioTableLocalPagination/StudioTableLocalPagination.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { forwardRef, useState } from 'react';
+import { StudioTableRemotePagination } from '../StudioTableRemotePagination';
+import type { Rows } from '../StudioTableRemotePagination';
+import { useTableSorting } from '../../hooks/useTableSorting';
+import { getRowsToRender } from '../StudioTableRemotePagination/utils';
+
+type StudioTableLocalPaginationProps = {
+  columns: Record<'accessor' | 'value', string>[];
+  rows: Rows;
+  size?: 'small' | 'medium' | 'large';
+  isSortable?: boolean;
+  pagination?: {
+    pageSizeOptions: number[];
+    pageSizeLabel: string;
+    nextButtonText: string;
+    previousButtonText: string;
+    itemLabel: (num: number) => string;
+  };
+};
+
+export const StudioTableLocalPagination = forwardRef<
+  HTMLTableElement,
+  StudioTableLocalPaginationProps
+>(({ columns, rows, isSortable = true, size = 'medium', pagination }, ref): React.ReactElement => {
+  const { pageSizeOptions, pageSizeLabel, itemLabel, nextButtonText, previousButtonText } =
+    pagination || {};
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [pageSize, setPageSize] = useState<number>(pagination ? pageSizeOptions[0] : undefined);
+
+  const { handleSorting, sortedRows } = useTableSorting(rows);
+  let rowsToRender: Rows;
+
+  if (isSortable) {
+    rowsToRender = getRowsToRender(currentPage, pageSize, sortedRows);
+  } else {
+    rowsToRender = getRowsToRender(currentPage, pageSize, rows);
+  }
```

Comment:
What do you think? Does it make the code easier to read like this? I assume sortedRows is undefined if they're not sorted?

By doing it this way, we do not need to assign the value to a let, but we can use the same constant as is.

```suggestion
   const rowsToRender: Rows = getRowsToRender(currentPage, pageSize, sortedRows || rows);
```

Reply:
Much easier to read. Fixed 😊